src/Blocks/BlockStems.h

```diff
@@ -58,25 +83,24 @@ class cBlockStemsHandler:
 	virtual int Grow(cChunk & a_Chunk, Vector3i a_RelPos, int a_NumStages = 1) override
 	{
 		auto oldMeta = a_Chunk.GetMeta(a_RelPos);
-		auto meta = oldMeta + a_NumStages;
-		a_Chunk.SetBlock(a_RelPos, m_BlockType, static_cast<NIBBLETYPE>(std::min(meta, 7)));  // Update the stem
-		if (meta > 7)
-		{
-			if (growProduce(a_Chunk, a_RelPos))
```

Comment:
Why have you removed the actual growth of the fruit from this function? Where has it gone...

Reply:
With the old function, when you right clicked with a bone meal on a full grown pumpkin / melon seed, a pumpkin / melon block spawn.
Here, it grows the plant to age 7 and doesn't spawn any blocks

Reply:
The problem is, that the `Grow()` function is also called on a normal block update. If it's removed here, then the fruit blocks will not spawn ever. So it needs to be put back here and a different function written for bonemeal growth that doesn't call `growProduce`.